src/Form/FormControl.js

```diff
@@ -0,0 +1,120 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+
+export const styleSheet = createStyleSheet('FormControl', () => {
+  return {
+    root: {
+      marginTop: 16,
+      display: 'flex',
+      flexDirection: 'column',
+      position: 'relative',
+    },
+    row: {
+      flexDirection: 'row',
+    },
+  };
+});
+
+/**
+ * FormControl
+ */
+export default class FormControl extends Component {
+  static propTypes = {
+    /**
+     * The contents of the `FormControl`.
+     */
+    children: PropTypes.node,
+    /**
+     * The CSS class name of the root element.
+     */
+    className: PropTypes.string,
+    /**
+     * Whether the label should be displayed in an error state.
+     */
+    error: PropTypes.bool,
+    required: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    error: false,
+    required: false,
+  };
+
+  static contextTypes = {
+    styleManager: PropTypes.object.isRequired,
+  };
+
+  static childContextTypes = {
+    formControl: PropTypes.object.isRequired,
+  };
+
+  state = {
+    dirty: false,
+    focused: false,
+  };
+
+  getChildContext() {
+    const { error, required } = this.props;
+    const { dirty, focused } = this.state;
+
+    return {
+      formControl: {
```

Comment:
should we prefix that key with `mui` to prevent conflict? Using a symbol would probably be too extrem.